### PR TITLE
[bitnami/thanos] Add compactor dataDir setting

### LINKED ISSUE #29835: [bitnami/thanos] data-dir should be configurable
### Name and Version

bitnami/thanos 15.7.27

### What is the problem this feature will solve?

I'm using Thanos compactor on an AWS EFS volume, shared between multiple k8s clusters and accounts.
That means I need to have a directory for each cluster, but the EFS controller doesn't allow to automatically create this directory, so I can't directly mount it on the compactor pod.

Mounting `/data` on the EFS root and setting `data-dir` to a subdirectory correctly creates a directory for each cluster, but the current chart value for `data-dir` is hard coded to `/data`, and setting it in `extraFlags` causes an error because the parameter can only be set once.

### What is the feature you are proposing to solve the problem?

The `data-dir` setting should be configurable, with `/data` as a default value.

### What alternatives have you considered?

I'm currently overriding the whole `args` list just to customize `data-dir`.

### PR DESCRIPTION
**Pull request by gdlx**
### Description of the change

Adds a `dataDir` compactor to customize the data directory.

### Benefits

Allows the setting of a different data directory without overriding the whole args list.

### Possible drawbacks

None

### Applicable issues

Fixes #29835

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution! Could you please bump the chart version in the _Chart.yaml_? This is necessary to test the changes and cut a new release.

In the same way, this is just a friendly reminder that signing your commits is important. Your signature certifies that you either authored the patch or have the necessary rights to contribute to the changes. You can find detailed information on how to do this in the [“Sign your work” section of our contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

Feel free to reach out if you have any questions or need assistance with the signing process.

**Comment by gdlx** [comment]
Hi @carrodher,
I just bumped the chart and fixed the markdown issue.
About the commit signature, it wasn't signed when opening the PR, but I think I fixed it before you asked. Did I do something wrong ?

**Comment by carrodher** [comment]
The DCO check is still failing, see https://github.com/bitnami/charts/pull/29856/checks?check_run_id=31396595990 for some tips about how to fix it

**Comment by gdlx** [comment]
@carrodher It seems to be all ok now.

File: bitnami/thanos/Chart.yaml [hunk 1]
```diff
@@ -35,4 +35,4 @@ maintainers:
 name: thanos
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/thanos
-version: 15.7.28
+version: 15.7.29
```
Comment (dgomezleon) [inline bitnami/thanos/Chart.yaml:38]:
```suggestion
version: 15.8.0
```

**Review by dgomezleon — COMMENTED** [review]
Thanks for contributing! Please, bump a minor version since it adds a new param.

**Review by dgomezleon — APPROVED** [review]

**Comment by gdlx** [comment]
@dgomezleon Is there something else needed for this review ?
It says 1 approving review is required to merge it but I'm not sure if it's all good.

**Review by dgomezleon — APPROVED** [review]

**Comment by gdlx** [comment]
Thanks !

**Pull request merged**